### PR TITLE
Remove bootstrap library fully with some minor fixes

### LINKED ISSUE #18756: Replace Bootstrap in Admin with Crayons
# User Story (Who / What / Why)
As a developer, I want a consistent styling approach on Admin and the main application so that it is easier to develop and there's consistent look and feel.

## Additional Context

At some point we were using a library called [Bootstrap](https://github.com/forem/forem/blob/1cb86b973ad5ee44438489c284b636ded60fc824/app/views/layouts/admin.html.erb#L20) to do the styling on the admin. However, we then built our own design system and started using that on the main application but didn't completely remove Bootstrap from the Admin app. As a result, it is still impacting some of the styles on the admin and causing inconsistent in preview and the end-user output.

Back in March, Pawel had tried to remove a ton of legacy Bootstrap dependant code from our admin app by converting it to use Crayons, however priorities changed and we had to switch gears. He has removed most of the bootstrap with a small portion of work still remaining to get rid of the entire library. 

More from the same issue: https://github.com/forem/forem/issues/18392#issuecomment-1238955086

This will involve some investigative work where we identify all the places that bootstrap is applying intentionally or unintentionally on the interface and updating to Crayons. After removing Bootstrap, we will need to go through the UI and ensure that we style with Crayons in the cases where Bootstrap was unintentionally styling the UI.

### Replace Bootstrap classes with Crayons and fix styling in admin

- [x] admin/badge_achievements
- [x] admin/badges
- [x] admin/broadcasts
- [x] admin/comments
- [x] admin/consumer_apps
- [x] admin/data_update_scripts
- [x] admin/feedback_messages
- [x] admin/listing_categories
- [x] admin/listings
- [x] admin/moderator_actions
- [x] admin/mods
- [x] admin/navigation_links
- [x] admin/organizations
- [x] admin/pages
- [x] admin/podcasts
- [x] admin/privileged_reactions
- [x] admin/profile_fields
- [x] admin/response_templates
- [x] admin/settings
- [x] admin/shared
- [x] admin/tags
- [x] admin/tools
- [x] admin/users
- [x] admin/welcome

### Final Cleanup
- [ ] Revisit all the PRs created for this issue, gather the glossary of bootstrap classes which were removed, and then recheck all those classes in entire codebase and update those too.
- [ ] Remove Bootstrap library

## QA Coordination:

- [ ] @anujbhavsar96 to participate in Design QA for this so loop in so we can QA on either Canary 1 or 2

### PR DESCRIPTION
**Pull request by rt4914**
## What type of PR is this? (check all applicable)

- [x] Refactor

## Description

This PR contains following types of changes:
- I revisited all the PRs that I had created for removing bootstrap and whatever issues were found are getting fixed in this PR.
- This PR overrides tags like `h1`, `h2`, `h3`, `h4`, `h5`, `h6`, `hr` and `a` so that the UI can match exactly same as that of with bootstrap library.
- Removes bootstrap library

## Related Tickets & Documents

- Related Issue #18756
- Closes #18756

### Replace `justify-content-` with `justify-`

Following classes have been replaced:
- `justify-content-start` with `justify-start`
- `justify-content-center` with `justify-center`
- `justify-content-end` with `justify-end`
- `justify-content-between` with `justify-between`
- `justify-content-around` with `justify-around`
- `justify-content-stretch` with `justify-stretch`

After removing bootstrap library:
| with justify-content-between | with justify-between |
| ------------- | ------------- |
| Screenshot 2023-01-11 at 1 04 59 PM | Screenshot 2023-01-11 at 1 04 51 PM |
| Screenshot 2023-01-11 at 1 06 52 PM | Screenshot 2023-01-11 at 1 06 41 PM | 

### Override heading tags

In below screenshot have a look at the width/height of the entire div to see that the heading tags properties have been fixed. 
| without Bootstrap Library  | with Bootstrap Library | After Overriding css |
| ------------- | ------------- | ------------- |
| Screenshot 2023-01-11 at 2 03 27 PM | Screenshot 2023-01-11 at 2 02 17 PM | Screenshot 2023-01-11 at 2 03 41 PM |
| height = 244.17 px | height = 207.97 px | height = 207.97 px |

### Override hr tag & a:hover tag

| without Bootstrap Library  | After Overriding css |
| ------------- | ------------- |
| Screenshot 2023-01-11 at 2 08 21 PM | Screenshot 2023-01-11 at 2 07 49 PM |
| Screenshot 2023-01-11 at 2 40 40 PM | Screenshot 2023-01-11 at 2 40 31 PM |

### Mistakes from previous PRs

**Spacing issue** PR #18855 
In this PR there was ` ` (space) missing before between tag `"` & `%>`. So replaced `"%>` with `" %>`

**Removed crayons-select class**
In `app/views/admin/consumer_apps/_form.html.erb` for `Platform selection` the `crayons-select` class was getting used for internal items also but was removed by mistake in https://github.com/forem/forem/pull/18861/files 
Add `crayons-select` to internal items too.

**Removed sr-only class based element by mistake**
In `app/views/admin/moderator_actions/index.html.erb` there was as label with class `sr-only` which was removed by mistake in https://github.com/forem/forem/pull/18880/files . Added that label back with `screen-reader-only` class.

### UI accessibility concerns?

No

## Added/updated tests?

- [ ] Yes
- [x] No

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [x] I will share this change internally with the appropriate teams

## [optional] Are there any post deployment tasks we need to perform?

@maestromac We will need to deploy this PR's changes on canary so that we can do Q/A.

## [optional] What gif best describes this PR or how it makes you feel?

Almost done

### CODE REVIEW

**Comment by rt4914** [comment]
@Ridhwana @lboogie04 Have a look at the PR description, I have added details/reasoning for every change in this PR.

Also, I need help in removing bootstrap library fully. Apart from changes in this PR, there places where bootstrap is still mentioned but I am not fully confident about how to remove them (check screenshot)

Screenshot 2023-01-11 at 2 46 38 PM

**Comment by gadkins** [comment]
Hi, @rt4914. I'm on the Uffizzi team providing pull request previews to forem. [This preview workflow](https://github.com/forem/forem/actions/runs/3891514544/jobs/6641833532) failed because we had forem on the wrong Uffizzi plan. Your project is now upgraded, so a preview for this PR should work if you re-run the workflow.

**Comment by Ridhwana** [comment]
> Also, I need help in removing bootstrap library fully. Apart from changes in this PR, there places where bootstrap is still mentioned but I am not fully confident about how to remove them (check screenshot)

Hey @rt4914 👋 I'll take a look at reviewing this PR today. I may not have a chance to follow up on the rest of the removal today as its a pretty meeting heavy day but I'll take a peek when I'm able to.  😊

**Review by Ridhwana — COMMENTED** [review]
Thanks @rt4914 ✨ Its so great to get to a point where we can remove the bootstrap library. Whoooop! 

I've added some thoughts to the PR on which I'd love to get your feedback.

File: app/views/admin/consumer_apps/_form.html.erb
Comment (rt4914) [reply app/views/admin/consumer_apps/_form.html.erb:9]: **Aded crayons-select class**
In `app/views/admin/consumer_apps/_form.html.erb` for `Platform selection` the `crayons-select` class was getting used for internal items also but was removed by mistake in https://github.com/forem/forem/pull/18861/files 
Added `crayons-select` to internal items too.

So basically this was the exact code (i.e. using crayons-select for internal items as well as for overall item) before I started removing bootstrap but by mistake in #18861 it was removed and I was using this class at only one place and therefore I have added this class back for internal-items.

Although, UI wise there is not change.

File: app/assets/stylesheets/admin.scss
Comment (rt4914) [reply app/assets/stylesheets/admin.scss:14]: I do agree with your overall reasoning that we should be using crayons-style everywhere in should not override h1-h6 tags in admin.scss file.

But this PR is purely focused on removing bootstrap only and as a I result I have not added those classes to h1-h6 tags.

Once we are done with removing bootstrap I think it would be much more clean that we then remove these h1-h6 tags too and instead use appropriate crayons-class.

My reason is that if we apply those changes directly in this PR then there might be introduction of some new crayon-classes and as a result might create confusion if someone revisits this PR in future.
Also, wanted to keep this PR simple, because if something breaks after merging it because of bootstrap removal, it would be easy to revert this PR and fix those issues.


Still, do let me know if apply those changes directly in this PR makes more sense, would be happy to fix them directly in here.

Comment (Ridhwana) [reply app/assets/stylesheets/admin.scss:14]: Based on some more investigation it seems that we already have the `crayons-title` and `crayons-subtitle-x` classes on most of the headings which was a nice surprise. 

However, I think what you've outlined above is fine, let's tackle this in a follow up PR instead  ✨

File: app/views/admin/consumer_apps/_form.html.erb
Comment (Ridhwana) [reply app/views/admin/consumer_apps/_form.html.erb:9]: @rt4914 apologies I'm not sure I'm following.
When both classes are in the code like the change you've made above, one of them get ignored in the HTML because they are duplicated. This is what the HTML looks like for:

```
 <%= select_tag :platform, options_for_select(ConsumerApp.platforms.invert, selected: @app.platform, class: "crayons-select"), class: "crayons-select", data: { "consumer-app-target" => "platform", "action" => "consumer-app#checkPlatform" } %>

```

Screenshot 2023-01-13 at 11 05 27

Hence, I would think we need to remove the duplicate class that is not getting applied:

```suggestion
    <%= select_tag :platform, options_for_select(ConsumerApp.platforms.invert, selected: @app.platform), class: "crayons-select", data: { "consumer-app-target" => "platform", "action" => "consumer-app#checkPlatform" } %>
```

The form renders fine and still applies the class correctly.

Screenshot 2023-01-13 at 11 02 03

Perhaps I'm misunderstanding something though?

File: app/assets/stylesheets/admin.scss
Comment (rt4914) [reply app/assets/stylesheets/admin.scss:14]: Great. Thanks.

File: app/views/admin/consumer_apps/_form.html.erb
Comment (rt4914) [reply app/views/admin/consumer_apps/_form.html.erb]: Great. Thanks a lot for detailed explanation. It means that before working of removing bootstrap this implementation was incorrect at that time too.

I have removed the inner crayons-select class to fix duplicacy.

**Review by Ridhwana — APPROVED** [review]
Thanks @rt4914 ✨

File: app/views/admin/consumer_apps/_form.html.erb
Comment (Ridhwana) [reply app/views/admin/consumer_apps/_form.html.erb]: It's a pleasure.

> It means that before working of removing bootstrap this implementation was incorrect at that time too.

Yip, exactly. Thank you for fixing it!

**Comment by rt4914** [comment]
@mirie @maestromac This PR is currently under Q/A in canary1. Should we wait for that to finish or merge this PR?

**Comment by rt4914** [comment]
@Ridhwana @lboogie04 I missed this earlier, any suggestions on this comment: https://github.com/forem/forem/pull/18945#issuecomment-1378480187

**Comment by Ridhwana** [comment]
> @Ridhwana @lboogie04 Have a look at the PR description, I have added details/reasoning for every change in this PR.
> 
> Also, I need help in removing bootstrap library fully. Apart from changes in this PR, there places where bootstrap is still mentioned but I am not fully confident about how to remove them (check screenshot)
> 
> Screenshot 2023-01-11 at 2 46 38 PM

@rt4914 the bootstrap referred to in `container-compose.yml`, `docker-compose.yml` and `docker-compose.uffizi.yml` refers to the [name of the parameter that we pass through the `entrypoint.sh` file](https://github.com/forem/forem/blob/fcf7c2f6ffe75272eaba16218dcc14f5dc3de441/scripts/entrypoint.sh#L40) further down in the list that allows us to run some custom command. 

The `codeql-analysis` is just a comment and the `yarn-1.22.18.js` reference seems to be an unrelated mime type. Hence, I don't think we need to remove any of these.

**Comment by rt4914** [comment]
> @rt4914 the bootstrap referred to in `container-compose.yml`, `docker-compose.yml` and `docker-compose.uffizi.yml` refers to the [name of the parameter that we pass through the `entrypoint.sh` file](https://github.com/forem/forem/blob/fcf7c2f6ffe75272eaba16218dcc14f5dc3de441/scripts/entrypoint.sh#L40) further down in the list that allows us to run some custom command.
> 
> The `codeql-analysis` is just a comment and the `yarn-1.22.18.js` reference seems to be an unrelated mime type. Hence, I don't think we need to remove any of these.

Thanks @Ridhwana

**Pull request merged**
